packages/material-ui-styles/src/makeStyles/makeStyles.d.ts

```diff
@@ -7,21 +7,13 @@ import {
 import { Omit } from '@material-ui/types';
 import { DefaultTheme } from '../defaultTheme';
 
-/**
- * `makeStyles` where the passed `styles` do not depend on props
- */
-export default function makeStyles<Theme = DefaultTheme, ClassKey extends string = string>(
-  style: Styles<Theme, {}, ClassKey>,
-  options?: Omit<WithStylesOptions<Theme>, 'withTheme'>
-): (props?: any) => ClassNameMap<ClassKey>;
-/**
- * `makeStyles` where the passed `styles` do depend on props
- */
 export default function makeStyles<
   Theme = DefaultTheme,
-  Props extends {} = {},
+  Props extends object = {},
```

Comment:
This change is a breaking one for me
```ts
// Type 'boolean' does not satisfy the constraint 'object'.ts(2344)
// ___________________________________🔻
const useStyles = makeStyles<Theme, boolean>(() =>
  createStyles({
    root: {
      display: (show) => (show ? 'block' : 'none'),
    }
  })
}
```

Reply:
Probably, I was using it not as it was designed but still, FYI if it was unintentional

Reply:
Related to https://github.com/mui-org/material-ui/issues/23706#issuecomment-733913529

Reply:
> Just wanted to request more caution around things like this being in a patch release even if you could argue people were using it wrong as a justification for why it's "not a breaking change".

Any type change is potentially backwards incompatible and can therefore be considered breaking. In the DefinitelyTyped community, as well as recommended by TypeScript core members, we don't bump SemVer major changes if patch types or add new features. Otherwise we couldn't apply any change without constantly bumping the major.